relays/bin-substrate/src/cli/encode_message.rs

```diff
@@ -56,41 +85,45 @@ pub trait CliEncodeMessage: Chain {
 }
 
 /// Encode message payload passed through CLI flags.
-pub(crate) fn encode_message<Source: Chain, Target: Chain>(
+pub(crate) fn encode_message<Source: CliEncodeMessage, Target: Chain>(
 	message: &Message,
 ) -> anyhow::Result<RawMessage> {
 	Ok(match message {
 		Message::Raw { ref data } => data.0.clone(),
 		Message::Sized { ref size } => {
-			let expected_xcm_size = match *size {
+			let destination = match Target::ID {
+				MILLAU_CHAIN_ID => CustomNetworkId::Millau.as_network_id(),
+				RIALTO_CHAIN_ID => CustomNetworkId::RialtoParachain.as_network_id(),
+				RIALTO_PARACHAIN_CHAIN_ID => CustomNetworkId::RialtoParachain.as_network_id(),
+				_ => return Err(anyhow::format_err!("Unsupported target chain: {:?}", Target::ID)),
+			};
+			let expected_size = match *size {
 				ExplicitOrMaximal::Explicit(size) => size,
 				ExplicitOrMaximal::Maximal => compute_maximal_message_size(
 					Source::max_extrinsic_size(),
 					Target::max_extrinsic_size(),
 				),
-			};
-
-			// there's no way to craft XCM of the given size - we'll be using `ExpectPallet`
-			// instruction, which has byte vector inside
-			let mut current_vec_size = expected_xcm_size;
-			let xcm = loop {
-				let xcm = xcm::VersionedXcm::<()>::V3(
-					vec![xcm::v3::Instruction::ExpectPallet {
-						index: 0,
-						name: vec![42; current_vec_size as usize],
-						module_name: vec![],
-						crate_major: 0,
-						min_crate_minor: 0,
-					}]
-					.into(),
-				);
-				if xcm.encode().len() <= expected_xcm_size as usize {
-					break xcm
-				}
-
-				current_vec_size -= 1;
-			};
-			xcm.encode()
+			} as usize;
+
+			let at_target_xcm = vec![xcm::v3::Instruction::ClearOrigin; expected_size].into();
+			let at_target_xcm_size =
+				Source::encode_wire_message(Target::ID, at_target_xcm)?.encoded_size();
+			let at_target_xcm_overhead = at_target_xcm_size.saturating_sub(expected_size);
```

Comment:
It'll probably make our messages 1 or 2 bytes above the maximal allowed size because `Compact<u32>` (length of XCM instructions vector) is larger for larger vectors => this message will be rejected. Current approach may made messages 1-byte smaller in edge cases, but it is still fine. And yeah - it seems a bit hacky, but I haven't found any better way

Reply:
Yes, agree that this version is the safest.